### ISSUE TITLE
Linter forcing me not to declare a Type in valid Typescript code

### ISSUE THREAD

**Issue by bradws** [palantir/tslint#1653]
### Bug Report

- __TSLint version__: `3.15.1`
- __TypeScript version__: `2.0.3`
- __Running TSLint via__: Node.js in the package.json file as a script

#### TypeScript code being linted

```ts
let myBool: boolean = true;
```

with `tslint.json` configuration:

```json
{
  "rules": {
    "class-name": true,
    "comment-format": [
      true,
      "check-space"
    ],
    "curly": true,
    "eofline": true,
    "forin": true,
    "indent": [
      true,
      "spaces"
    ],
    "label-position": true,
    "label-undefined": true,
    "max-line-length": [
      true,
      140
    ],
    "member-access": false,
    "member-ordering": [
      true,
      "static-before-instance",
      "variables-before-functions"
    ],
    "no-arg": true,
    "no-bitwise": true,
    "no-console": [
      true,
      "debug",
      "info",
      "time",
      "timeEnd",
      "trace"
    ],
    "no-construct": true,
    "no-debugger": true,
    "no-duplicate-key": true,
    "no-duplicate-variable": true,
    "no-empty": false,
    "no-eval": true,
    "no-inferrable-types": true,
    "no-shadowed-variable": true,
    "no-string-literal": false,
    "no-switch-case-fall-through": true,
    "no-trailing-whitespace": true,
    "no-unused-expression": true,
    "no-unused-variable": true,
    "no-unreachable": true,
    "no-use-before-declare": true,
    "no-var-keyword": true,
    "object-literal-sort-keys": false,
    "one-line": [
      true,
      "check-open-brace",
      "check-catch",
      "check-else",
      "check-whitespace"
    ],
    "quotemark": [
      true,
      "single"
    ],
    "radix": true,
    "semicolon": [
      "always"
    ],
    "triple-equals": [
      true,
      "allow-null-check"
    ],
    "typedef-whitespace": [
      true,
      {
        "call-signature": "nospace",
        "index-signature": "nospace",
        "parameter": "nospace",
        "property-declaration": "nospace",
        "variable-declaration": "nospace"
      }
    ],
    "variable-name": false,
    "whitespace": [
      true,
      "check-branch",
      "check-decl",
      "check-operator",
      "check-separator",
      "check-type"
    ]
  }
}

```

#### Actual behavior

When I have the following Typescript line in a test spec:
```ts
let myBool: boolean = true;
```
the linter complains with:

(no-inferrable-types) app/app.component.spec.ts[39, 17]: LHS type (boolean) inferred by RHS expression, remove type annotation

So I change the code to read:
```ts
let myBool = true;
```
and the error goes away.

Why is this? Shouldn't I be able to write valid Typescript in a spec?

Note this problem does not occur in a regular Typescript class. My spec file seems to transpile just fine.

This doesn't mean to be a nit-picking question: I am just curious.
Thanks.

**Comment by IllusionMH** [comment]
There is [`no-inferrable-types`](https://palantir.github.io/tslint/rules/no-inferrable-types/) in your `tslint.json` which warns you about such cases with primitives (number, string, boolean).
Just remove it from your configuration ant there will be no warnings.

To make sure that definitions are required you can check [`typedef`](https://palantir.github.io/tslint/rules/typedef/) rule and add it to your `tslint.json`

**Comment by bradws** [comment]
@IllusionMH Thank-you

**Comment by shoff** [comment]
That REALLY should NOT be a default.

**Comment by imKiruSrinivas** [comment]
> number, string, boolean



> There is [`no-inferrable-types`](https://palantir.github.io/tslint/rules/no-inferrable-types/) in your `tslint.json` which warns you about such cases with primitives (number, string, boolean).
> Just remove it from your configuration ant there will be no warnings.
> 
> To make sure that definitions are required you can check [`typedef`](https://palantir.github.io/tslint/rules/typedef/) rule and add it to your `tslint.json`

Just courious to check, if [`no-inferrable-types`] is removed then it will not warn not only for boolean but also for number, string. Will it be feasible solution?
